### PR TITLE
Implement service workers

### PR DESCRIPTION
**Pull request by casperisfine**
Ref: https://github.com/Shopify/pitchfork/issues/110

It's not uncommon for applications to want to have some background threads doing regular work for various purposes (e.g. emitting metrics, prewarming node local caches, etc).

Currently the best way to do it, is to spawn some background threads in the `after_mold_fork`, but this isn't great as the mold may fork at any moment to spawn a new worker, and forking while background threads may be doing work is risky.

Instead we can provide a dedicated service worker process to run these threads.

### CODE REVIEW

File: docs/CONFIGURATION.md [hunk 1]
```diff
@@ -422,6 +422,39 @@ after_request_complete do |server, worker, env|
 end
 ```
 
+### `before_service_worker_ready` (experimental)
+
+Experimental and may change at any point.
+
+If defined, Pitchfork will spawn one extra worker, called a service worker
+which doesn't accept incomming requests, but allows to perform service tasks
```
Comment (paarthmadan) [inline docs/CONFIGURATION.md:430]: Had to scratch my Ruby itch, sorry for the drive-by review 😅

```suggestion
which doesn't accept incoming requests, but allows to perform service tasks
```

Comment (casperisfine) [reply docs/CONFIGURATION.md:430]: ❤️
